.circleci/config.yml

```diff
@@ -0,0 +1,103 @@
+version: 2.1
+
+executors:
+  default-executor:
+    docker:
+      - image: "cimg/node:lts"
+    working_directory: ~/project
+    resource_class: medium
+commands:
+  restore_test_cache:
+    steps:
+      - restore_cache:
+          keys:
+            - v1-deps-{{ .Environment.CIRCLE_JOB }}-{{ .Branch }}-{{ .Revision }}
+            - v1-deps-{{ .Environment.CIRCLE_JOB }}-{{ .Branch }}
+            - v1-deps-
+  save_test_cache:
+    steps:
+      - save_cache:
+          key: v1-deps-{{ .Environment.CIRCLE_JOB }}-{{ .Branch }}-{{ .Revision }}
+          paths:
+            - node_modules
+  setup:
+    steps:
+      - run:
+          name: Setup
+          command: |
+            npm --production=false install
+            npm --production=false update
+            npm --production=false prune
+  test:
+    steps:
+      - run:
+          name: Test
+          command: |
+            npm run test
+  setup_deploy:
+    steps:
+      - run:
+          name: Setup Deploy
+          command: |
+            echo "export NPM_TAG=test" >> $BASH_ENV
+            echo "export NODE_ENV=production" >> $BASH_ENV
+            echo "export RELEASE_TIMESTAMP=$(date +'%Y%m%d%H%M%S')" >> $BASH_ENV
+            npm run build
+            sudo chmod +x .circleci/before_deploy.sh
```

Comment:
Same comments here as with `scratch-render`:

Since this is the only place it's used, it seems like it would make sense to just move the `before_deploy.sh` contents into this script. Unless there's a reason not to do that...?

If it does need to be in a different file, I'd recommend doing a `chmod +x before_deploy.sh` and then `git commit` that, so you don't need the `chmod` step here.

Reply:
Move into main file!